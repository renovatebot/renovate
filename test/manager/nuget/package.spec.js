const { getPackageUpdates } = require('../../../lib/manager/nuget/package');
const nugetApi = require('../../../lib/datasource/nuget');
const defaultConfig = require('../../../lib/config/defaults').getConfig();

nugetApi.getNuspec = jest.fn();
nugetApi.getVersions = jest.fn();

describe('lib/manager/nuget/package', () => {
  describe('getPackageUpdates()', () => {
    let config;
    beforeEach(() => {
      config = {
        ...defaultConfig,
        depName: 'some-dep',
        currentVersion: '2.3.0',
        lineNumber: 1337,
        ignoreUnstable: true,
      };
    });
    it('returns empty if no versions are found', async () => {
      expect(await getPackageUpdates(config)).toEqual([]);
    });
    it('returns empty if current version is not semver', async () => {
      config.currentVersion = undefined;
      nugetApi.getVersions.mockReturnValueOnce(['1.0.0']);
      expect(await getPackageUpdates(config)).toEqual([]);
    });
    it('returns empty if latest version is not semver', async () => {
      nugetApi.getVersions.mockReturnValueOnce(['5.0.0.0']);
      expect(await getPackageUpdates(config)).toEqual([]);
    });
    it('returns empty if highest version is current version', async () => {
      nugetApi.getVersions.mockReturnValueOnce([
        '1.0.0',
        config.currentVersion,
      ]);
      expect(await getPackageUpdates(config)).toEqual([]);
    });
    it('returns higher version if available', async () => {
      nugetApi.getVersions.mockReturnValueOnce(['1.0.0', '2.3.1', '2.4.0']);
      expect(await getPackageUpdates(config)).toMatchSnapshot();
    });
    it('ignores unstable version if specified', async () => {
      nugetApi.getVersions.mockReturnValueOnce(['3.0.0-alpha1']);
      expect(await getPackageUpdates(config)).toEqual([]);
    });
  });
});
