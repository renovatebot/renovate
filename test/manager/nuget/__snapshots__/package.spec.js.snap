// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lib/manager/nuget/package getPackageUpdates() returns higher version if available 1`] = `
Array [
  Object {
    "changeLogFromVersion": "2.3.0",
    "changeLogToVersion": "2.4.0",
    "lineNumber": 1337,
    "newVersion": "2.4.0",
    "newVersionMajor": 2,
    "newVersionMinor": 4,
    "type": "minor",
  },
]
`;
